CMakeLists.txt

```diff
@@ -1,5 +1,6 @@
 PROJECT( tomahawk )
 CMAKE_MINIMUM_REQUIRED( VERSION 2.8.6 )
+set(CMAKE_EXPORT_COMPILE_COMMANDS "ON")
```

Comment:
This is first commit in this pr. This can be used with things like https://github.com/Valloric/YouCompleteMe , which is a completion/etc plugin for Vim. 
See http://www.reddit.com/r/vim/comments/2aaoyr/youcompleteme_with_cmake_build_system/